src/Microsoft.AspNetCore.Authentication/AuthenticationSchemeOptions.cs

```diff
@@ -36,5 +36,15 @@ public virtual void Validate()
         /// If set, will be used as the service type to get the Events instance instead of the property.
         /// </summary>
         public Type EventsType { get; set; }
+
+        /// <summary>
+        /// Used to ensure that the options are only initialized once.
+        /// </summary>
+        public bool Initialized { get; set; }
```

Comment:
I took a look at the options changes, and there's indeed nothing (yet) that could help with that. Have you considered introducing an `IOptionsValidator<T>`/`IOptionsInitializerT>` that would be run after all the `IConfigureOptions<T>`? (or you could bring back ordered setups).